### PR TITLE
[BACKLOG-14407] null mask values were causing a XmiParsing error. Now…

### PR DESCRIPTION
**Pull request by DFieldFL**
… only adding mask values if they are not null.

### CODE REVIEW

**Comment by DFieldFL** [comment]
@pentaho/ackbar and @nantunes could you please review. The XmiParsing error that we are seeing is coming from #135

File: src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java [hunk 1]
```diff
@@ -30,9 +30,9 @@
 import org.pentaho.metadata.model.SqlPhysicalTable;
 
 import java.util.List;
+import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.junit.Assert.*;
```
Comment (nantunes) [inline src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java]: The current checkstyle rules don't check static import statements, but I was told it was best to still follow the rule of not using `*` imports... Not that it matters. 😋

File: src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java [hunk 2]
```diff
@@ -66,10 +66,18 @@ public void testImportTableCanExcludeColumns() throws Exception {
       importTableDefinition( database, "aSchema", "aTable", "klingon", importStrategy );
     List<IPhysicalColumn> physicalColumns = physicalTable.getPhysicalColumns();
     assertEquals( 3, physicalColumns.size() );
-    assertContainsColumn( "quantityordered", physicalColumns );
-    assertContainsColumn( "productcode", physicalColumns );
-    assertContainsColumn( "status", physicalColumns );
 
+    // Depending on the class the formatting values are set or unset. If NULL formatting values were set on the
+    // property map, this would cause an XmiParsing Error. Testing that values that are NULL are not set in the
+    // property map.
```
Comment (nantunes) [inline src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java]: Where is it testing the NULL case?

Comment (DFieldFL) [inline src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java]: It really depends on what ValueMeta type that is used. In this test I have listed what is null and then not put into the properties map. I am doing a very generic test ensures that ensures there are no nulls being set for the those formatting values that were added.

ValueMetaInteger (quantityordered) has currencySymbol set to Null
ValueMetaString (productcode and status) has mask and currencySymbol set to Null

Comment (nantunes) [inline src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java]: Sorry, only not I saw `assertNotNullPropertyValue` returns `true` **only** if the key is set.

File: src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java (hunk 1 above)
Comment (nantunes) [inline src/test/java/org/pentaho/metadata/automodel/PhysicalTableImporterTest.java]: Yeah... I most rules we only get the **what* not the **why**, so it's harder to generalize in new cases. If you have that impression and have been using it, i'm ok with it.

**Comment by DFieldFL** [comment]
:+1:

**Pull request merged**